src/ui/scroll-window.js

```diff
@@ -12,93 +32,153 @@ export class ScrollWindow extends Phaser.Scene {
         this.characterData = characterData;
     }
 
+    addObject(object) {
+        this.data.objects.push(object);
+    }
+
     // Objects passed in need the setX, setY, setVisible functions available
-    addObjects(x, y, width, numColumns, objects) {
-        if (objects.length > 0) {
-            // Add scroll bar and button
-            let scrollBar = this.add
-                .image(width, 0, "scroll-bar")
-                .setDepth(2)
-                .setOrigin(0, 0);
-            let scrollHeaderHeight = 16;
-            let scrollButton = this.add
-                .image(width, scrollHeaderHeight, "scroll-button")
-                .setDepth(2)
-                .setOrigin(0, 0);
+    addObjects(data) {
+        if (data.objects.length > 0) {
+            this.refresh(data);
+        }
+    }
+
+    refresh(data = this.data) {
```

Comment:
Do you have plans to be able to click and drag the scroll button?

Reply:
Yes, I made a card for it but didn’t think it was very high priority. I’ve already done drag for the world map though so maybe if I refer back to that I can get it done today in this PR

Reply:
Alright, you can now drag it and use the top/bottom buttons to move it as well.